Added Invoice Related definitions to Glossary (#323)
killbill/killbill-docs: userguide/glossary/Kill-Bill-Glossary.adoc

[[Bundle]]Bundle:: A collection of subscriptions associated with a single service instance. For example, a customer might have a base subscription for her phone, plus an add-on subscription for international dialing. Because both subscriptions are associated with the same phone, they are collected into a single bundle. An account may have several bundles, each with one or more subscriptions. Also known as a "subscription bundle."

> Bill Cycle Day:: The day of the month the invoice is created for an account. Note that this is applicable only to monthly and quarterly subscriptions.

Reviewer: `monthly and quarterly subscriptions.` -> to monthly based subscriptions (`MONTHLY`, `QUARTERLY`, `ANNUAL`, `BIMESTRIAL`, `BIANNUAL`, `BIENNIAL`) as opposed to (`DAILY`, `THIRTY_DAYS`, `WEEKLY`, ..)

Author: Fixed

== I

Invoice:: An invoice is a bill provided to a customer for charges that are payable on the customer's account. A single invoice consolidates charges for all subscriptions held by that customer for a specified time period. Invoices may be set to be paid automatically, or may be paid manually by the customer. An invoice consists of one or more invoice items.

> Invoice Item:: Represents a single item charged on an invoice. Given an active subscription, one could see multiple items for that subscription on a single invoice including recurring items, usage items, fixed price items, etc. There can also be items for different subscriptions on the same invoice, as well as items that are unrelated to subscriptions such as adjustments and taxes. 

Reviewer 2: ```suggestion
Invoice Item:: Represents a single item charged on an invoice. Given an active subscription, one could see multiple items for that subscription on a single invoice, including recurring items, usage items, fixed price items, etc. There can also be items for different subscriptions on the same invoice, as well as items that are unrelated to subscriptions, such as adjustments and taxes. 
```

Author: Fixed

> Invoice Item:: Represents a single item charged on an invoice. Given an active subscription, one could see multiple items for that subscription on a single invoice including recurring items, usage items, fixed price items, etc. There can also be items for different subscriptions on the same invoice, as well as items that are unrelated to subscriptions such as adjustments and taxes. 

Reviewer: 👏

Author: Copied these from the invoice slate docs: https://killbill.github.io/slate/#invoice 😀

> Invoice Item Type:: Specifies the type of invoice item. Can be one of `EXTERNAL_CHARGE`, `FIXED`, `RECURRING`, `REPAIR_ADJ`, `CBA_ADJ`, `ITEM_ADJ`, `USAGE`, `TAX`, `PARENT_SUMMARY`. Refer to the

Reviewer 2: ```suggestion
Invoice Item Type:: Specifies the type of invoice item, which can be one of `EXTERNAL_CHARGE`, `FIXED`, `RECURRING`, `REPAIR_ADJ`, `CBA_ADJ`, `ITEM_ADJ`, `USAGE`, `TAX`, `PARENT_SUMMARY`. Refer to the
```

Author: Fixed

> Invoice Item Type:: Specifies the type of invoice item. Can be one of `EXTERNAL_CHARGE`, `FIXED`, `RECURRING`, `REPAIR_ADJ`, `CBA_ADJ`, `ITEM_ADJ`, `USAGE`, `TAX`, `PARENT_SUMMARY`. Refer to the

Reviewer: Is it intended to have two 'columns' `::` on all these?

Author: Yes, I think, I just copied @FlowWriter's style 😀  I tried removing the double colon (`::`)  and found that the document is not displayed properly.

Reviewer 2: Right - this is the Asciidoctor syntax for glossary items. https://docs.asciidoctor.org/asciidoc/latest/sections/glossary/

Author: 👍

> https://docs.killbill.io/latest/userguide_subscription.html#components-invoicing[_Subscription Billing_] document to know more. 

Reviewer 2: ```suggestion
https://docs.killbill.io/latest/userguide_subscription.html#components-invoicing[_Subscription Billing_] document for more information. 
```

Author: Fixed

Invoice Payment:: Payment made against a specific invoice. By default, Kill Bill attempts to make a payment against an invoice using the account's default payment method. 

> Invoice Refund:: Refund amount associated with an invoice. Kill Bill allows issuing a full or partial refund against an invoice payment. 

Reviewer: I think the following would make more sense: `Refund amount associated with an Invoice Payment.` Refunds are against payment, and the terminology `Invoice Refund` really means `(Invoice Payment) Refund`. Happy to hear other suggestions...

Author: Changed to `Refund amount associated with an invoice payment`. If anyone has any other suggestions, I can make changes accordingly.

Reviewer 2: That sounds more accurate.

Author: @FlowWriter, are you simply agreeing with @sbrossie here or is there an action item for me? 😀

> Invoice Amount:: The amount that a customer is charged as part of an invoice. Also known as "Invoice Total",  "Total",  "Charged Amount" or simply "Amount". At a high level, it is calculated by adding the amounts on all the chargeable invoice items associated with an invoice. See https://docs.killbill.io/latest/invoice_examples.html#_charged_amount[charged amount] to know more.

Reviewer 2: ```suggestion
Invoice Amount:: The amount that a customer is charged as part of an invoice. Also known as "Invoice Total",  "Total",  "Charged Amount" or simply "Amount". At a high level, it is calculated by adding the amounts on all the chargeable invoice items associated with an invoice. See https://docs.killbill.io/latest/invoice_examples.html#_charged_amount[charged amount] for more information.
```

Author: Fixed

> Invoice Balance:: The amount that the customer owes as part of an invoice. It is positive if the customer owes something or zero otherwise. At a high level,  it is calculated by subtracting the paid amount from the charged amount. See https://docs.killbill.io/latest/invoice_examples.html#_invoice_balance[invoice balance] to know more.

Reviewer: `from the charged amount` -> `from the Invoice Amount` (to reuse preceding definition). Don't know about the uppercase, this is a question for our expert in house @FlowWriter

Author: Changed to `from the invoice amount`. @FlowWriter, could really use your suggestion with regards to capitalization. So, if a previously defined term is used in another definition, should it be capitalized?

Reviewer 2: I do not capitalize glossary term within the glossary (or anywhere else for that matter). The grammar rule is that only proper nouns are capitalized (Reno, January, Big Foot). Also, I find it too difficult to remember to consistenly capitalize terms. And finally, it looks a little odd with all those capitalized words sprinkled across a document. However, you can link to them within the glossary. I think there's an example elsewhere in there you can use. You have to set up the anchor next to the glossary term and then reference the anchor.

> Invoice Credit:: The amount that we owe to a customer as part of an invoice.

Reviewer 2: ```suggestion
Invoice Credit:: The amount that your business owes to a customer as part of an invoice.
```

Reviewer 2: This could benefit from an example that explains why an amount might be credited on an invoice (like incorrectly billed or customer contested amount).

Author: Made suggested changes and added an example, let me know how it looks

Reviewer 2: Looks good!

> Invoice Status:: Status of an invoice. Can be one of `DRAFT`, `COMMITTED` or `VOID`.

Reviewer 2: ```suggestion
Invoice Status:: The current state of an invoice, which can be in a `DRAFT`, `COMMITTED` or `VOID` state.
```

Author: Fixed

[[Parent]]Parent:: An account that contains one or more child accounts. Setting up a parent-child associations lets you define which entity is responsible for paying the invoice. See also <<Child>>.

> Payment Method:: A Kill Bill record that represents an abstraction corresponding to a payment scheme like a credit card, debit card, or PayPal. An account can have multiple payment methods corresponding to different payment gateways. For example, an account can have a credit card payment method as well as a Paypal payment method.Each account also has a *default* payment method associated with it. When the account needs to be charged for recurring payments, the default payment method is used. 

Reviewer 2: ```suggestion
Payment Method:: A Kill Bill record that represents an abstraction corresponding to a payment scheme like a credit card, debit card, or PayPal. An account can have multiple payment methods corresponding to different payment gateways. For example, an account can have a credit card payment method as well as a PayPal payment method. Each account also has a *default* payment method associated with it. When the account needs to be charged for recurring payments, the system uses the default payment method. 
```

Author: Fixed

> Payment::  A Kill Bill record that encapsulates charging an end-user against a payment method. It has a state that indicates the status of the underlying transaction.

Reviewer: I think what you have for `Payment` is more related to a `Payment Transaction`, so we could first define this, and then define `Payment` as a collection of `Payment Transaction` (e.g `authorization`, `refund`, ..) using the same `Payment Method`.

Author: Added `Payment Transaction`, `Payment Attempt` definitions and updated `Payment` definition based on what is specified in the [Payment slate docs](https://killbill.github.io/slate/#payment).

Reviewer 2: It all looks good.

> Payment::  A Kill Bill record that encapsulates charging an end-user against a payment method. It has a state that indicates the status of the underlying transaction.

Author: I've changed the definition for `Payment`, let me know how the new definition looks

> Payment::  A Kill Bill record that encapsulates charging an end-user against a payment method. It has a state that indicates the status of the underlying transaction.

Reviewer 2: Very nice!

> Payment::  A Kill Bill record that encapsulates charging an end-user against a payment method. It has a state that indicates the status of the underlying transaction.

Reviewer 2: Oops. Forgot to mention... check the order of the terms here. "Payment" should come before "Payment Attempt," which should come before "Payment Method," etc.

> Payment::  A Kill Bill record that encapsulates charging an end-user against a payment method. It has a state that indicates the status of the underlying transaction.

Author: Fixed
